src/guest_memory.rs

```diff
@@ -646,10 +651,11 @@ pub trait GuestMemory {
                 Ok(0) => return Ok(total),
                 // made some progress
                 Ok(len) => {
-                    total += len;
-                    if total == count {
-                        break;
-                    }
+                    total = match total.checked_add(len) {
+                        Some(x) if x == count => break,
+                        Some(x) if x < count => x,
+                        _ => return Err(Error::CallbackOutOfRange),
+                    };
```

Comment:
In the case of `x == count`, we still need to update `total` before breaking out of the loop, as `total` will be used to report how many bytes were transferred, and without updating it the final transfer chunk will not be considered. That is why the CI is failing. 
```suggestion
                    match total.checked_add(len) {
                        Some(x) if x <= count => {
                            total = x;
                            if total == count {
                                break
                            }
                        }
                        _ => return Err(Error::CallbackOutOfRange)
                    }
```